vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -262,7 +262,7 @@ def forward_hpu(
         if hasattr(self, "scaling_factors") or hasattr(
                 self, "scaling_factor") or self.sin is None:
             self.prepare_cos_sin(positions, offsets)
-        num_tokens = positions.shape[0] * positions.shape[1]
+        num_tokens = positions.numel()
```

Comment:
any specific reason to update this?

Reply:
The `positions` of Deepseek-R1 is a flattened 1D tensor.
Using `numel()` is much robust and aligned with `forward_native()`:
https://github.com/HabanaAI/vllm-fork/blob/deb019dcf767d012434f1f301547c95fdbf7be9a/vllm/model_executor/layers/rotary_embedding.py#L175-L176